torch/_refs/nn/functional/__init__.py

```diff
@@ -376,3 +378,37 @@ def gelu(a: TensorLikeType, approximate: str = "none") -> TensorLikeType:
         return a * 0.5 * (1 + torch.erf(a * kAlpha))
     else:
         raise RuntimeError("approximate argument must be either none or tanh.")
+
+
+def prelu(a: TensorLikeType, weight: TensorLikeType) -> TensorLikeType:
+    """
+    Reference implementation of torch.nn.functional.prelu
+    """
+    check(
+        isinstance(a, TensorLike),
+        lambda: f"prelu: Expected `a` to be tensor, but got: {type(a)}",
+    )
+    check(
+        isinstance(weight, TensorLike),
+        lambda: f"prelu: Expected `weight` to be tensor, but got: {type(weight)}",
+    )
+
+    if weight.numel() != 1:
+        check(a.ndim > 0, lambda: "Not allow zero-dim input tensor.")
+        channel_size = a.shape[1] if a.ndim >= 2 else 1
+        check(
+            weight.numel() == channel_size,
+            lambda: f"Mismatch of parameter numbers and input channel size. Found parameter numbers ="
+            f" {weight.numel()} and channel size = {channel_size}.",
+        )
+
+    check(
+        weight.ndim == 0 or weight.ndim == 1,
+        lambda: f"prelu: Expected `weight` to be a scalar or 1D tensor, but got: "
+        f"ndim = {weight.ndim}",
+    )
+    weight = prims.broadcast_in_dim(
```

Comment:
Instead of calling the broadcast_in_dim prim directly (which looks totally correct, by the way), should we instead call broadcast_to here? 

https://github.com/pytorch/pytorch/blob/459090e3ceaff7276a4fbf51161e3a8e627a275f/torch/_refs/__init__.py#L1680

Generally we want to prefer calling the most specific reference implementation for the behavior we're interested in.

Reply:
I see. Not sure if `broadast_to` works when `a.ndim > 2` though:
```
>>> import torch
>>> a = torch.ones(3, 2, 3)
>>> b = torch.ones(2)
>>> torch.prelu(a, b)
tensor([[[1., 1., 1.],
         [1., 1., 1.]],

        [[1., 1., 1.],
         [1., 1., 1.]],

        [[1., 1., 1.],
         [1., 1., 1.]]])
>>> torch.broadcast_to(b, (3, 2, 3))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: The expanded size of the tensor (3) must match the existing size (2) at non-singleton dimension 2.  Target sizes: [3, 2, 3].  Tensor sizes: [2]
>>> 
```

Reply:
Wait, why does prelu(a,b) work - b is not broadcastable to a, it should error out.

Reply:
Bonus points if we can make the docs more readable, @soulitzer